Fix node digging grouops and sounds

Emojigit opened the pull request:
This PR fixes the digging groups and sound by having them follow their recipe node. To make sure the data is readable during registration, all material mods are added as optional dependencies.

This PR is ready for review.

Emojigit committed:
Fix node digging grouops and sounds

This PR fixes the digging groups and sound by having them follow their recipe node. To make sure the data is readable during registration, all material mods are added as optional dependencies.

Emojigit committed:
Remove debug code

SmallJoker commented on shapes.lua:
```diff
@@ -224,8 +265,8 @@ function facade.register_corbel(modname, subname, recipeitem, desc)
 		is_ground_content = false,
-		groups = {cracky = 3, oddly_breakable_by_hand = 2, stone = 1},
-		sounds = default.node_sound_stone_defaults(),
+		groups = get_dig_groups(recipeitem),
```
What about the `stone` group? I believe that one should be preserved.

Emojigit replied:
`stone` is not a dig group, but rather a group identifying that it is stone.

SmallJoker replied:
I know. Is there a particular reason why that group is no longer used? It can still be relevant for recipes.

SmallJoker commented on shapes.lua:
```diff
@@ -2,6 +2,47 @@
+	-- This node don't want to be dug
+	if not item.groups then
+		return {}
+	end
```
This field is never `nil` due to the `core.nodedef_default` metatable.
This field is never `false` due to `luaL_checktype(L, index, LUA_TTABLE);` in `c_content.cpp`.

The check can be removed. In a sane environment, this table guaranteed to exist.

Emojigit committed:
Fix node digging grouops and sounds

This PR fixes the digging groups and sound by having them follow their
recipe node. To make sure the data is readable during registration, all
material mods are added as optional dependencies.
